src/domainobjects/instrument.py

```diff
@@ -1,61 +1,139 @@
+        """
+
+        cache = Cache()
+
         records = []
-        current_tickers = {}
-
-        for i in range(0, record_count):            
-            asset_class = self.generate_asset_class()         
-            ticker = self.generate_ticker()
-            coi = self.generate_coi()
-            exchange_code = self.\
-                generate_sequential_exchange_code(current_tickers, ticker)
-            cusip = str(self.generate_random_integer(length=9))
-            isin = self.generate_isin(coi, cusip)
-            ric = self.generate_ric(ticker, exchange_code)
-            sedol = self.generate_random_integer(length=7)
-            j = i+1
-
-            records.append({
-                'instrument_id': j,
+
+        for i in range(start_id, start_id+record_count):
+            record = self.generate_record(i, cache)
+            records.append(record)
+            self.persist_record(
+                [record['ric'], str(record['cusip']), str(record['isin'])]
+            )
+
+        self.persist_records("instruments")
+        return records
+
+    def generate_record(self, id, cache):
+        """ Generate a single instrument
+
+        Parameters
+        ----------
+        id : int
+            Current id of the instrument to generate, used as a pseudo
+            exchange code to ensure uniquely generated instruments
+        cache : dict
+            Storeage medium for tickers and countries of issuance.
+
+        Returns
+        -------
+        dict
+            A single back office position object
+        """
+
+        asset_class = self.generate_asset_class()
+        ticker = self.generate_ticker(cache)
+        coi = self.generate_coi(cache)
+        exchange_code = id
+        cusip = self.generate_random_integer(length=9)
+        isin = self.generate_isin(coi, cusip)
+        ric = self.generate_ric(ticker, exchange_code)
+        sedol = self.generate_random_integer(length=7)
+        return {
+                'instrument_id': id,
                 'ric': ric,
                 'isin': isin,
                 'sedol': sedol,
                 'ticker': ticker,
                 'cusip': cusip,
                 'asset_class': asset_class,
                 'coi': coi,
-                'time_stamp': datetime.now()})
-
-            database.persist("instruments", [ric, cusip, isin])
+                'time_stamp': datetime.now()
+            }
 
-            if (j % int(records_per_file) == 0):
-                self.write_to_file(file_num, records)
-                file_num += 1
-                records = []
-        
-        if records != []: 
-            self.write_to_file(file_num, records)
-        current_tickers = {}
+    def generate_asset_class(self):
+        """ Generate a predetermined asset class for instruments
 
-        database.commit_changes()
+        Returns
+        -------
+        String
+            Asset class of an instrument will be 'Stock'
+        """
```

Comment:
Potentially change to 'Equity'. 

Find out if we want other classes such as :

Bond, Fund, Option, Future, Warrant